build-tools/packages/build-cli/src/commands/check/policy.ts

```diff
@@ -160,97 +179,89 @@ export class CheckPolicy extends BaseCommand<typeof CheckPolicy> {
-
-			if (pipeString !== undefined) {
-				try {
-					pipeString
-						.split("\n")
-						.map((line: string) =>
-							this.handleLine(
-								line,
-								pathRegex,
-								exclusions,
-								handlersToRun,
-								handlerExclusions,
-							),
-						);
-				} finally {
-					try {
-						runPolicyCheck(handlersToRun, this.flags.fix);
-					} finally {
-						this.logStats();
-					}
-				}
+			const stdInput = await readStdin();
+
+			if (stdInput !== undefined) {
+				filePathsToCheck.push(...stdInput.split("\n"));
 			}
+		} else {
+			const repo = new Repository({ baseDir: gitRoot });
+			const gitFiles = await repo.gitClient.raw(
+				"ls-files",
+				"-co",
+				"--exclude-standard",
+				"--full-name",
+			);
 
-			return;
+			filePathsToCheck.push(...gitFiles.split("\n"));
 		}
 
-		CheckPolicy.pathToGitRoot = childProcess
-			.execSync("git rev-parse --show-cdup", { encoding: "utf8" })
-			.trim();
-
-		const p = childProcess.spawn("git", [
-			"ls-files",
-			"-co",
-			"--exclude-standard",
-			"--full-name",
-		]);
-
-		let scriptOutput = "";
-		p.stdout.on("data", (data) => {
-			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
-			scriptOutput = `${scriptOutput}${data.toString()}`;
-		});
-		p.stdout.on("close", () => {
+		const commandContext: CheckPolicyCommandContext = {
+			pathRegex,
+			exclusions,
+			handlers: handlersToRun,
+			handlerExclusions,
+			gitRoot,
+			context,
+		};
+
+		await this.executePolicy(filePathsToCheck, commandContext);
+	}
+
+	private async executePolicy(
+		pathsToCheck: string[],
+		commandContext: CheckPolicyCommandContext,
+	): Promise<void> {
+		try {
+			pathsToCheck.map((line: string) => this.handleLine(line, commandContext));
+		} finally {
 			try {
-				scriptOutput
-					.split("\n")
-					.map((line: string) =>
-						this.handleLine(
-							line,
-							pathRegex,
-							exclusions,
-							handlersToRun,
-							handlerExclusions,
-						),
-					);
+				runPolicyCheck(commandContext, this.flags.fix);
 			} finally {
-				try {
-					runPolicyCheck(handlersToRun, this.flags.fix);
-				} finally {
-					this.logStats();
-				}
+				this.logStats();
 			}
-		});
+		}
 	}
 
-	// route files to their handlers by regex testing their full paths
-	// synchronize output, exit code, and resolve decision for all handlers
-	routeToHandlers(file: string, handlers: Handler[], handlerExclusions: HandlerExclusions): void {
+	/**
+	 * Routes files to their handlers by regex testing their full paths. Synchronize the output, exit code, and resolve
+	 * decision for all handlers.
+	 */
+	private routeToHandlers(file: string, commandContext: CheckPolicyCommandContext): void {
+		const { context, handlers, handlerExclusions, gitRoot } = commandContext;
+
+		// Use the repo-relative path so that regexes that specify string start (^) will match repo paths.
+		const relPath = context.repo.relativeToRepo(file);
+
```

Comment:
Haven't debugged why, but this change limits at least the package.json match policies to root package.json.

Reply:
On Windows this will produce a Windows path using backslashes which will fail handlers only expecting forward slashes.